### PR TITLE
API Endpoint to Modify Experts

### PR DESCRIPTION
**Pull request by dynamitechetan**
Fixes issue #258 
This Endpoint accepts 5 parameters. model,group,language,expert,content.

Screenshots for the change: 
<table><th>If Successfully Created</th><th>If file does not exists</th>
<tr><td></td><td></td></table>

### CODE REVIEW

File: src/ai/susi/server/api/cms/ModifyExpertService.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        File expert = new File(language, expert_name + ".txt");
+        // Checking for file existence
+        if(!expert.exists()){
+            JSONObject error = new JSONObject();
+            error.put("response","Expert Does Not Exists");
+            return new ServiceResponse(error);
+        }
+        // Reading Content for expert
+        String content = call.get("content", "");
+        if(expert_name==null){
```
Comment (saurabhjn76) [inline src/ai/susi/server/api/cms/ModifyExpertService.java]: Shouldn't this be 'content==null' ?

Comment (dynamitechetan) [inline src/ai/susi/server/api/cms/ModifyExpertService.java]: Oh, I forgot this one. I will make the changes. Thanks.

**Comment by dynamitechetan** [comment]
Made the changes, please review.

File: src/ai/susi/server/api/cms/ModifyExpertService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        if(Objects.equals(content, "")){
+            JSONObject error = new JSONObject();
+            error.put("response","expert content not given");
+            return new ServiceResponse(error);
+        }
+        // Writing to File
+        try (FileWriter file = new FileWriter(expert)) {
+            file.write(content);
+            JSONObject success = new JSONObject();
+            success.put("response",expert.getName()+" modified successfully");
```
Comment (Orbiter) [inline src/ai/susi/server/api/cms/ModifyExpertService.java]: please add:
success.put("accepted", true);
to make processing in client easier

File: src/ai/susi/server/api/cms/ModifyExpertService.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        if(!expert.exists()){
+            JSONObject error = new JSONObject();
+            error.put("response","Expert Does Not Exists");
+            return new ServiceResponse(error);
+        }
+        // Reading Content for expert
+        String content = call.get("content", "");
+        if(Objects.equals(content, "")){
+            JSONObject error = new JSONObject();
+            error.put("response","expert content not given");
```
Comment (Orbiter) [inline src/ai/susi/server/api/cms/ModifyExpertService.java]: please add:
error.put("accepted", false);
to make processing in client easier

File: src/ai/susi/server/api/cms/ModifyExpertService.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        String group_name = call.get("group", "knowledge");
+        File group = new File(model, group_name);
+        String language_name = call.get("language", "en");
+        File language = new File(group, language_name);
+        String expert_name = call.get("expert", null);
+        File expert = new File(language, expert_name + ".txt");
+        // Checking for file existence
+        if(!expert.exists()){
+            JSONObject error = new JSONObject();
+            error.put("response","Expert Does Not Exists");
```
Comment (Orbiter) [inline src/ai/susi/server/api/cms/ModifyExpertService.java]: please add:
error.put("accepted", false);
to make processing in client easier

File: src/ai/susi/server/api/cms/ModifyExpertService.java (hunk 2 above)
Comment (dynamitechetan) [inline src/ai/susi/server/api/cms/ModifyExpertService.java]: okay 👍

**Comment by dynamitechetan** [comment]
@orbiter made changes, please review

**Pull request merged**